### DISCUSSION TITLE
Error 15 - Detected by Imperva

### DISCUSSION THREAD

**Discussion by JacobHobday** [ultrafunkamsterdam/undetected-chromedriver#1211] [Q&A]
I have an application that automatically reserves driving tests on:
https://driverpracticaltest.dvsa.gov.uk/
I can get the login page fine however once the bot has put in the reference and license number to login Imperva detects the bot and returns error 15.
Any suggestions?
I've tried:
-Converted the program to UC from just pure selenium so UC gets the page and selenium is used in tandem.
-Disabling headless
-Loading default chrome profile dir
-Using random_user_agent to generate user agent string

Here are the web driver options:
options = webdriver.ChromeOptions()
options.add_argument('start-maximized')
options.add_argument('disable-infobars')
options.add_argument('--disable-extensions')
options.add_argument('--disable-gpu')
options.add_argument('--disable-dev-shm-usage')
options.add_argument('--no-sandbox')
options.add_argument('--ignore-certificate-errors')
options.add_argument('--allow-running-insecure-content')
options.add_argument('--disable-popup-blocking')
options.add_argument(f'user-agent={userAgent}')
options.add_argument('--disable-blink-features=AutomationControlled')
options.add_argument("--disable-logging")
options.add_argument(r'--user-data-dir=C:\Users\jacob\AppData\Local\Google\Chrome\User Data\Default')
#options.add_argument("−−incognito")
#options.headless = HEADLESS
options.add_argument("--disable-default-apps")
options.add_argument("--disable-notifications")
options.add_argument("--disable-plugins-discovery")
options.add_argument("--disable-hang-monitor")
options.add_argument("--disable-prompt-on-repost")

**Comment by jiwaCh** [comment]
Did you mange to solve the issue?

**Reply by JacobHobday** [reply]
No, I tried zenrows too without any luck.

**Reply by jiwaCh** [reply]
okay, thanks for letting me know

**Comment by luqman-hussain** [comment]
Hi @JacobHobday have you had any luck with this or found another approach?

**Comment by SafeerAbbas624** [comment]
change this 
options = webdriver.ChromeOptions()
to this
 options = uc.ChromeOptions()

**Comment by ultrafunkamsterdam** [comment]
Or read README and remove most of your custom options as they will be passed twice otherwise
